components/admin_console/license_settings/modals/upload_license_modal.tsx

```diff
@@ -0,0 +1,281 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {useRef} from 'react';
+import {useSelector, useDispatch} from 'react-redux';
+
+import {FormattedDate, FormattedMessage} from 'react-intl';
+
+import {DispatchFunc} from 'mattermost-redux/types/actions';
+import {uploadLicense} from 'mattermost-redux/actions/admin';
+import {getLicenseConfig} from 'mattermost-redux/actions/general';
+import {getLicense} from 'mattermost-redux/selectors/entities/general';
+import {ClientLicense} from 'mattermost-redux/types/config';
+import {getCurrentLocale} from 'selectors/i18n';
+
+import {GlobalState} from 'types/store';
+
+import {isModalOpen} from 'selectors/views/modals';
+
+import GenericModal from 'components/generic_modal';
+import WomanArmOnTable from 'components/common/svg_images_components/woman_arm_on_table_svg';
+import HandsSvg from 'components/common/svg_images_components/hands_svg';
+import FileSvg from 'components/common/svg_images_components/file_svg';
+import LoadingWrapper from 'components/widgets/loading/loading_wrapper';
+
+import {ModalIdentifiers} from 'utils/constants';
+
+import {closeModal} from 'actions/views/modals';
+
+import {localizeMessage} from 'utils/utils';
+import {getMonthLong} from 'utils/i18n';
+
+import './upload_license_modal.scss';
+
+type Props = {
+    onExited?: () => void;
+}
+
+const UploadLicenseModal: React.FC<Props> = (props: Props): JSX.Element | null => {
+    const dispatch = useDispatch<DispatchFunc>();
+
+    const [fileObj, setFileObj] = React.useState<File | null>(null);
+    const [isUploading, setIsUploading] = React.useState(false);
+    const [serverError, setServerError] = React.useState<string | null>(null);
+    const fileInputRef = useRef<HTMLInputElement>(null);
+
+    const currentLicense: ClientLicense = useSelector(getLicense);
+    const locale = useSelector(getCurrentLocale);
+
+    const handleChange = () => {
+        const element = fileInputRef.current;
+        if (element === null || element.files === null || element.files.length === 0) {
+            return;
+        }
+        setFileObj(element.files[0]);
```

Comment:
Should this do a pre-check on the file size that it's `> 0`? The user shouldn't have empty files with a `.mattermost-license` prefix, but if they did, they could see odd UI. The only case I can think of is they rename an empty file to the license name on accident, or they have a network failure right at the beginning of downloading the license.
![empty-license](https://user-images.githubusercontent.com/13738432/148122249-868b1b5b-584e-4dce-8d6b-6564135ceab0.png)

Reply:
sure! it is an extreme scenario but worth keeping it into account, I like defensive programming, specially if the validation is cheap in terms of performance.